src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -119,7 +120,7 @@ private static Version GetICUVersion()
             }
```

Comment:
Instead of IsWindows check, we can just use presence/absence of the GetICUVersion internal method to tell whether the ICU is present (future proofing for when ICU is enabled on Windows); and return null 0.0.0.0 when it is absent.

Reply:
Interesting we have this check in the first place in .Unix file. I agree we should have a method telling if ICU is used or not. That what we are going to have anyway when we start enabling ICU for Windows.

---
In reply to: [379152410](https://github.com/dotnet/runtime/pull/32132#discussion_r379152410) [](ancestors = 379152410)

Reply:
We have the Windows Detection on a .Unix file because TestUtilities is not cross compiled by OS, so the convention is to use runtime checks instead.